src/natcap/invest/ndr/ndr.py

```diff
@@ -996,6 +990,42 @@ def _mult_op(*array_nodata_list):
         gdal.GDT_Float32, target_nodata)
 
 
+def _sum_rasters(raster_path_list, target_nodata, target_result_path):
+    """Sum the rasters in `raster_path_list`.
+
+    Args:
+        raster_path_list (list): list of single band raster paths.
+        target_nodata (float): desired target nodata value.
+        target_result_path (string): path to float 32 target raster
+            multiplied where all rasters are not nodata.
+
+    Returns:
+        None.
+
+    """
+    def _sum_op(*array_nodata_list):
+        """Multiply non-nodata stacks."""
+        result = numpy.empty(array_nodata_list[0].shape)
+        result[:] = target_nodata
+        valid_mask = numpy.full(result.shape, True)
+        for array, nodata in zip(*[iter(array_nodata_list)]*2):
+            if nodata is not None:
+                valid_mask &= ~numpy.isclose(array, nodata)
+        result[valid_mask] = array_nodata_list[0][valid_mask]
+        for array in array_nodata_list[2::2]:
```

Comment:
Passing in the nodata values as raw and indexing this way is a neat implementation!

Reply:
Is there a reason to grab the first array into result instead of iterating with `array_nodata_list[0::2]`?

Reply:
I refactored all this - see above